python_modules/dagster/dagster_tests/core_tests/structured_config_tests/test_basic_structured_config.py

```diff
@@ -351,3 +354,88 @@ def pipeline_requires_config():
 
     with pytest.raises(DagsterInvalidConfigError):
         validate_run_config(pipeline_requires_config)
+
+
+@pytest.mark.skipif(sys.version_info < (3, 8), reason="requires python3.8")
+def test_cached_property():
+    from functools import cached_property
+
+    counts = {
+        "plus": 0,
+        "mult": 0,
+    }
+
+    class SomeConfig(Config):
+        x: int
+        y: int
+
+        @cached_property
+        def plus(self):
+            counts["plus"] += 1
+            return self.x + self.y
+
+        @cached_property
+        def mult(self):
+            counts["mult"] += 1
+            return self.x * self.y
+
+    config = SomeConfig(x=3, y=5)
+
+    assert counts["plus"] == 0
+    assert counts["mult"] == 0
+
+    assert config.plus == 8
+
+    assert counts["plus"] == 1
+    assert counts["mult"] == 0
+
+    assert config.plus == 8
+
+    assert counts["plus"] == 1
+    assert counts["mult"] == 0
+
+    assert config.mult == 15
+
+    assert counts["plus"] == 1
+    assert counts["mult"] == 1
+
+
+def test_cached_method():
```

Comment:
Python 3.9 also introduces [`functools.cache`](https://docs.python.org/3/library/functools.html#functools.cache) - wonder if this would also work out of the box (seems functionally quite similar to `cached_method`)

Reply:
It does not. 

```
    @pytest.mark.skipif(sys.version_info < (3, 9), reason="requires python3.9")
    def test_functools_cache():
        from functools import cache

        counts = {
            "plus": 0,
            "mult": 0,
        }

        class SomeConfig(Config):
            x: int
            y: int

            @cache
            def plus(self):
                counts["plus"] += 1
                return self.x + self.y

            @cache
            def mult(self):
                counts["mult"] += 1
                return self.x * self.y

        config = SomeConfig(x=3, y=5)

        assert counts["plus"] == 0
        assert counts["mult"] == 0

>       assert config.plus() == 8
E       TypeError: test_functools_cache.<locals>.SomeConfig.plus() missing 1 required positional argument: 'self'
```